subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/DefaultJavaPluginConvention.java

```diff
@@ -93,7 +101,8 @@ private File getReportsDir() {
 
     @Override
     public JavaVersion getSourceCompatibility() {
-        return srcCompat != null ? srcCompat : JavaVersion.current();
+        Provider<Integer> release = findRelease();
```

Comment:
:o: This could benefit from an extracted method to reuse in `getTargetCompatibility`